rust/zypp-agama-example/src/main.rs

```diff
@@ -53,12 +53,12 @@ fn main() -> Result<(), zypp_agama::ZyppError> {
     let default_root = "/".to_owned();
     let root = args.get(1).unwrap_or(&default_root);
 
-    zypp_agama::init_target(root, |text, step, total| {
+    let z = zypp_agama::init_target(root, |text, step, total| {
         println!("Initializing target: {}/{} - {}", step, total, text)
     })?;
 
     println!("List of existing repositories:");
-    let repos = zypp_agama::list_repositories()?;
+    let repos = zypp_agama::list_repositories(&z)?;
```

Comment:
well, if it is needed in all parts, that probably `z.list_repositories` makes more sense from my POV. This way we can also implement a bit of validation logic in rust ( like ensure that repo manager is created before calling its methods and so on ).